### PR TITLE
Add Colab notebook showing Hugging Face import via Turbine.

### PR DESCRIPTION
**Pull request by ScottTodd**
Preview: https://colab.research.google.com/github/scotttodd/iree/blob/colab-huggingface/samples/colab/pytorch_huggingface_whisper.ipynb

This shows a minimal import from Hugging Face using the style of code from https://github.com/nod-ai/SHARK-TestSuite/tree/main/e2eshark/pytorch/models. I'd like to expand on this with other advanced export APIs (e.g. using parameter files like Safetensors/GGUF/IRPA, Turbine-LLM, and custom kernels), but figured I'd start with something fairly well tested and simple.

The test case I forked from also used a text prompt for an audio -> text model, so this just shows some numbers as output... I'd like to show text or images as output, though that needs some interfacing (WhisperProcessor wraps WhisperFeatureExtractor and WhisperTokenizer, for example).

### CODE REVIEW

**Comment by ScottTodd** [comment]
@zjgarvey as you wrote https://github.com/nod-ai/SHARK-TestSuite/pull/171 recently, would you like to review this?

File: samples/colab/pytorch_huggingface_whisper.ipynb [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+        "##### Copyright 2024 The IREE Authors"
+      ],
+      "metadata": {
+        "id": "UUXnh11hA75x"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "#@title Licensed under the Apache License v2.0 with LLVM Exceptions.\n",
```
Comment (zjgarvey) [inline samples/colab/pytorch_huggingface_whisper.ipynb:33]: What does the "@"title do? Are there some ipynb viewers that render this as an actual title? At least in github's preview feature, it just leave the comment as-is.

**Review by zjgarvey — APPROVED** [review]
Yeah, this is really nice.

I didn't know iree and SHARK-Turbine got rolled together into a package, this is quite convenient.

File: samples/colab/pytorch_huggingface_whisper.ipynb (hunk 1 above)
Comment (ScottTodd) [reply samples/colab/pytorch_huggingface_whisper.ipynb:33]: Yeah these are designed for (interactive) viewing in Colab (https://colab.google/). Anyone can open a notebook from the URL and click "Runtime -> Run All" (or Connect -> run individually) to play with the code. There are a few alternatives to Colab for online Jupyter notebooks with different hosting tiers and features, but I'm most familiar with Colab.

Colab renders a single line of markdown following `#@title` above the cell:
image

You can also collapse cells that have titles: 
image

Could replace those titles embedded into code cells with text cells, but I like the direct link between the title and the code, as well as that collapsing ability. Maybe I'm overusing titles though. Just looked at a few notebooks from JAX and didn't see them in use there.

Comment (ScottTodd) [reply samples/colab/pytorch_huggingface_whisper.ipynb:33]: Switched to text cells.

Comment (zjgarvey) [reply samples/colab/pytorch_huggingface_whisper.ipynb:33]: Oh, I didn't mean it as a hard suggestion. I just haven't used google collab before. I guess it's nice that the github preview now has the titles. 

I noticed that boxes 1, 2, and 5 still have the inlined markdown, so if you want to be consistent with the change, I would modify those, too.

Comment (ScottTodd) [reply samples/colab/pytorch_huggingface_whisper.ipynb:33]: I hid those cells (by default) but kept the titles: 
image

e.g. copyright hidden vs fully expanded:
image
image

not really expecting people to focus on those sections, so I want the collapsing behavior

**Comment by ScottTodd** [comment]
> I didn't know iree and SHARK-Turbine got rolled together into a package, this is quite convenient.

Turbine already included IREE, but we're in the process of changing the branding / promoting it from shark-turbine to just iree-turbine: https://groups.google.com/g/iree-discuss/c/Bk58qwhaPEU/m/NwD90RwoAAAJ

**Pull request merged**